tt_metal/llrt/tt_cluster.cpp

```diff
@@ -583,6 +691,71 @@ void Cluster::initialize_ethernet_sockets() {
     }
 }
 
+void Cluster::reserve_ethernet_cores_for_tunneling() {
```

Comment:
a few high level comments in front of the bigger routines would help w/ understanding the flow: what do you have to search through?  what criteria is being looked for